### DISCUSSION TITLE
How to estimate energy demand?

### DISCUSSION THREAD

**Discussion by buyanqifande** [iiasa/message_ix#472] [MESSAGE modeling] [answered]
Hello! I am interested in message model and learnt that the energy demand are exogeneous in the message model. But how could it distinguished the difference between energy demand in different climate target? the emission section in the message model can give upper bound for carbon emission, but how to determine the input energy demand at very first stage?
Hope to get your help! Many thanks!

**Comment by behnam-zakeri** [comment, marked as answer]
Hello @buyanqifande. Thanks for the question. As you mentioned "demand" in MESSAGE is exogenous. So, the user initially defines demand for different energy services, e.g., demand for electricity, heat, etc. in a given sector like buildings etc. Then, the model finds the optimal mix of supply technologies that can meet this demand. Climate targets can be defined independent of demand. For example, you can add a carbon neutrality target for 2050 or define a cumulative carbon budget for the whole 21st Century. For determining input energy demand in future decades you can use projections for a specific country/region from other studies, or you can estimate demand based on projections of GDP and populations. You can refer to the [Shared Socio-economic Pathway (SSP)](https://tntcat.iiasa.ac.at/SspDb/dsd?Action=htmlpage&page=10) database for GDP and population projections.

**Reply by buyanqifande** [reply]
Thanks a lot! It really helps!!! 
I have an another follow up question. If we got the optimal mix of supply technologies for single energy services, say electricity. How to generate the primary energy consumption that are required by electricity demand? Does it relate to the conversion efficiency and capacity factor of electricity generation technologies? For instance, if we got the results that it should be xxx GW wind power capacity installed in country A,  then we can multiply the annual utilization hours with capacity to get the power generation.  And then the primary consumption of wind can be got by dividing the power generation by the conversion factor? 
Looking forward to your reply! Thanks!

**Reply by behnam-zakeri** [reply]
As you mentioned primary energy needed to meet a certain energy service, here electricity, depends on conversion inefficiencies and losses throughout the system, for bringing a fuel like coal to electricity. Yes, as you mentioned, electricity generation (in unit of energy) can be divided by conversion factors to calculate primary energy. For example, for 3 TWh electricity from a coal power plant, you may need 10 TWh of coal as the primary fuel. For renewables, e.g., wind energy, there are two ways to calculate primary energy: one is direct calculation and another is the substitution method. For more info, please see [here](https://www.worldenergydata.org/energy-accounting/#:~:text=The%20substitution%20method%20is%20performed,generated%20by%20each%20non%2Dcombustible.).

**Reply by buyanqifande** [reply]
Hello! Dear friends! @behnam-zakeri When I run the tutorial model, I find it hard to understand the meaning of unit "GWa" when it represents both capacity and activity in results part. From my understanding, activity is like electricity generation when we focus on electricity system. Does it means the  electricity volume that the 1 GW infrastructure make for a 8760 h (a year), that is 8760 GWh equal to GWa? For the capacity, does it means the active capacity for a certain period of time?  I try to find reference, but hardly find one   :(
Hope to hear from you! Many thanks!

**Reply by LauWien** [reply]
Dear @buyanqifande,
Many thanks for reaching out!
You understood correctly, GWa is a unit of energy and activity and it shows a capacity of 1 GW being active in one year. SO, as you mentioned 1 GWa = 8760 GWh. The MESSAGEix formulation is agnostic to units. It means the user can choose a proper unit depending on their field and the scope of their work. And there is no unit conversion in the MESSAGEix formulation. In our tutorials, we have chosen unit of GWa for activity and GW for capacity. This makes parameterization convenient, e.g., in adding values for "input" and "output" parameters. I hope this helps.

**Reply by buyanqifande** [reply]
Hello! I am struggle with fossil energy supply these days. I found there is a westeros_fossil_resource file that show how to add fossil resoure on primary level and link the primary energy with secondary energy. However, since the model also state that there could be grade "a" and "b" of coal. How can we get the value of grade "a" and "b" as the output after the model solve? By the way, I find it hard to understand the last fig in the model file, which title is "westeros energy system extraction".  The plot contains both "y" and "coal" in the bar plot. What's the meaning of "y"? From my understanding, there should be a optimization of supply source base on whether cost or resource potential constraint? So the picture I imgine we can get is how many coal of grade a and grade b is needed in order to meet the secondary demand of coal plant generation. Maybe it has similarity with technology choice progress since electricity demand can be met by different electricity generation technologies. But in terms of primary energy, we may want to know for instance how many oil/gas/coal are from different sources (whether different resource types or from different regions). So in that sense, it's helpful to get the value of different types, that is the value of grade a and b in the previous example. 
Hope to get you help! Many thanks! 谢谢！
![fig1](https://user-images.githubusercontent.com/25165236/194084164-12717728-cbdc-48c0-b007-f505a235c62c.PNG)
